### PR TITLE
Adds cataloger for Azure blob storage

### LINKED ISSUE #995: Add catalog method to azure storage
**Is your feature request related to a problem? Please describe.**
With #955 a new catalog endpoint has been introduced.
#961 was opened to add missing storages and mongo is on its way to be merged.

This issue is for implementing the cataloger interface in Azure

**Describe the solution you'd like**
Implement the cataloger in azure.

### PR DESCRIPTION
**Pull request by gowdabas**
**What is the problem I am trying to address?**
This PR implements the catalog method for Azure blob storage provider.
Also, upgraded the azblob pkg version to v0.7.0 because we need  version >= v0.4.0 for ListBlobs API to pass in the token.

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**
Fixes #995

### CODE REVIEW

**Comment by arschles** [comment]
I've asked @gowdabas to file this pull request, so that I could more easily debug the e2e failure. I am getting ready for GopherCon today, so if anyone else has ideas on what's going on here, that of course would be most welcome!

**Comment by gowdabas** [comment]
@arschles No problem at all. I have fixed the formatting issues and looks like the build is all good now. Thank you !

File: pkg/storage/azureblob/cataloger.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+	"github.com/gomods/athens/pkg/paths"
+	"strings"
```
Comment (marpio) [inline pkg/storage/azureblob/cataloger.go]: Can you format the imports? I think go-import can do that. std lib first, empty line and other pkgs.

Comment (gowdabas) [reply pkg/storage/azureblob/cataloger.go]: Thanks @marpio . Fixed it.

**Comment by manugupt1** [comment]
looks like all the changes have been adressed. merging this. thanks 💯

**Comment by gowdabas** [comment]
@manugupt1 Thank you 👍

**Pull request merged**